171234526 damage on impact

Adrian-de-Lucas-Gomez opened the pull request:
[Finishes #171234526] Los impactos ya funcionan y se aplican los daños (quedará para la fase de testing los ajustes pertinentes). También se han añadido a Resources las imágenes de algunos de los objetos que ya tenemos.

Adrian-de-Lucas-Gomez committed:
Prueba con los rangos de la fuerza de impacto

Se han probado distintos rangos para la fuerza del impacto calculada de la velocidad lineal escalada con la masa y simplificado a la longitud de la fuerza. Por ahora se está calculando sobre el jugador pero deberia de ser desde el objeto con mayor velocidad, diria yo

Adrian-de-Lucas-Gomez committed:
Mas pulido al daño por impacto

Se ha puesto un prop en escena para comprobar que al recibir un impacto lo suficientemente fuerte se aplique daño al jugador en medida de la velocidad y masa del objeto. Tambien se han metido imagenes de las armas arojadizas en Resources.

Adrian-de-Lucas-Gomez committed:
Totalmente funcional

Funciona tanto con el impacto de un objeto pesado como de otro jugador si lleva suficiente velocidad. Se ha hecho un método auxiliar para evitar repetir código.

AndresRuizB commented on src/SourceFiles/CollisionHandler.cpp:
```diff
@@ -1,4 +1,18 @@
+
+	if (impact >= 50 && impact < 65) {playerHealth->subtractLife(2);}
+
+	if (impact >= 65) {playerHealth->subtractLife(3);}
```
los limites de daño mejor en constants.h

Adrian-de-Lucas-Gomez replied:
Ya está, lo acabo de poner.

AndresRuizB reviewed (changes requested):
Lo de mover los limites de daño a constants

Adrian-de-Lucas-Gomez committed:
Daños pasados a constantes

Se han realizado el cambio pedido  para tener los parametros de daño en Constants.h

AndresRuizB reviewed (approved):
Perf ;D

Adrian-de-Lucas-Gomez committed:
Daños bien puestos xD

Se me escapó

EstebenR reviewed (changes requested):
La colision no revisa si es un jugador, en la linea 56 del CollisionHandler.cpp el if solo tiene un '=' y no un '==' por tanto no estaba haciendo la comprobación xd.

Hay que:
*Hacer que sí que sea una comparación.
*Añadir el userData al collider pasandole el Entity al que va pegado cuando se crea.
*Hacer una comprobacion pidiendo el componente del user data (GetUserData().GetComponent o el equivalente que esté hecho para pedir el componente de una entidad) mirando si tiene health para poder restarsela

Adrian-de-Lucas-Gomez committed:
Solucionadas las peticiones de Esteban

Ahora se le pasa al collider la entidad como UserData para poder acceder a lo que queramos. Aqui lo que cogeremos es el componente Health si es un jugador para poder restar puntos de salud en caso de que sea necesario.

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 171234526-DamageOnImpact

Adrian-de-Lucas-Gomez committed:
Merge con master arreglado y todo funcional

Ya todo esta listo para meterlo en la rama master

AdriAlvarez00 reviewed (approved):
Correcto, pero igual habría que implementar una forma de diferenciar entre objetos duros y blandos
